research/picai/preprocessing_transforms.py

```diff
@@ -0,0 +1,176 @@
+from typing import Optional, Sequence, Tuple, Union
+
+import numpy as np
+import SimpleITK as sitk
+
+
+def resample_img(
+    image: sitk.Image,
+    spacing: Sequence[float] = (2.0, 2.0, 2.0),
+    size: Optional[Sequence[int]] = None,
+    is_label: bool = False,
+    pad_value: Optional[Union[float, int]] = 0.0,
+) -> sitk.Image:
+    """
+    Resample images to target resolution spacing.
+
+    Adapted from https://github.com/DIAGNijmegen/picai_prep.
+
+    Args:
+        image (sitk.Image): Image to be resized.
+        spacing (Sequence[float]): Spacing between voxels along each dimentsion.
+        size (Sequence[int]): Target size in voxels (z, y, x).
+        is_label (bool): Whether or not this is an annotation.
+        pad_value (Optional[Union[float, int]]): Amount of padding to use.
+
+    Returns:
+        sitk.Image: The resampled image.
+    """
+    # get original spacing and size
+    original_spacing = image.GetSpacing()
+    original_size = image.GetSize()
+
+    # convert PICAI z, y, x convention to SimpleITK's convention
```

Comment:
Based on the reversal below, it's worth noting that SimpleITKs convention is x, y, z?